### DISCUSSION TITLE
Override name generated from generic superclass

### DISCUSSION THREAD

**Discussion by va1erian** [smallrye/smallrye-graphql#1986] [Q&A]
Hello, we have a output type POJO that inherits from a generic super class, something like:
`class Foo extends Entity<Bar>`

is there a way to prevent smallrye from generating a graphql type of name Foo_Bar? Using the name annotation only overrides the name before the underscore.

**Comment by phillip-kruger** [comment]
Have you tried the `@Name` annotation ? Although that might just replace Foo with the name defined in the annotation. I don't think that is possible at the moment. @jmartisk ??

**Reply by jmartisk** [reply]
Hm. Why should `@Name` override just the first part of the name, and not the whole name? That sounds a bit like a bug, no?

**Reply by phillip-kruger** [reply]
Well, what would happen if you add a name to `Bar` ? Both of the elements have names (maybe from the annotation, maybe from the classname. The fact that it is Foo_Bar is because of generics, and to distinguish between other generic type. I don't think it's a but, we just need to find a clean way to do this.

**Reply by jmartisk** [reply]
If you specify a name for `Bar`, you specify a name for `Bar`, not for `Foo<Bar>` - TBH I don't quite understand why we should include the name of `Bar` in the name of `Foo` at all. What exactly does it help to distinguish? The value of the type parameter might not have any effect on the GraphQL fields created from it.

If we had a field whose type is `Entity<Bar>` directly, then that's different, but here it's `Foo`, which is a unique name that doesn't need further distinction, no?

Maybe I'm just forgetting some important detail, but this is how it looks to me

**Reply by phillip-kruger** [reply]
And what if you also have a query that return Bar (without Foo wrapping it ) ? Or Foo<Bla> ? we need to make this unique.

**Reply by jmartisk** [reply]
Then that query would return Bar, not Foo. I don't see the problem.

`Entity<Bar>` -> `Entity_Bar`
`Foo extends Entity<Bar>` -> `Foo`
`Bar` -> `Bar`

**Reply by va1erian** [reply]
Thanks for your replies. I agree with @jmartisk,  Foo is no longer a generic class so there is no need imho to append a type parameter name. the `@Name` annotation doesn't stop this behavior, 
`
@Name("Test")
 class Foo extends Entity<Bar>
 `
 gives Test_Bar

**Reply by phillip-kruger** [reply]
I'll leave this up to @jmartisk to investigate.

**Reply by jmartisk** [reply]
I tend to think we should change this, I'm just not completely sure about the consequences in terms of the breaking change. But probably it shouldn't be that bad given that if people want the old name, they can override it with `@Name`.. @mskacelik do you have an opinion on this?

**Reply by mskacelik** [reply]
I also think having the `_EXTENSIONCLASS` name part for the class is unnecessary since the `Foo` is already a "unique identifier" in the schema. :)
